src/Gambit/SubscriptionGambit.php

```diff
@@ -26,14 +26,12 @@ protected function conditions(AbstractSearch $search, array $matches, $negate)
     {
         $actor = $search->getActor();
 
-        // might be better as `id IN (subquery)`?
-        $method = $negate ? 'whereNotExists' : 'whereExists';
-        $search->getQuery()->$method(function ($query) use ($actor, $matches) {
-            $query->selectRaw('1')
-                  ->from('discussion_user')
-                  ->whereColumn('discussions.id', 'discussion_id')
-                  ->where('user_id', $actor->id)
-                  ->where('subscription', $matches[1] === 'follow' ? 'follow' : 'ignore');
+        $method = $negate ? 'whereNotIn' : 'whereIn';
+        $search->getQuery()->$method('id', function ($query) use ($actor, $matches) {
+            $query->selectRaw('discussion_id')
```

Comment:
Just one small thing: since we're selecting a normal column here, this can use `select()`. `selectRaw()` was only needed in the old version, because `1` was an SQL expression, not a column identifier.